cmake/SundialsBuildOptionsPre.cmake

```
@@ -292,3 +292,7 @@ sundials_option(SUNDIALS_TEST_DEVTESTS BOOL
 # Include unit tests in regression tests
 sundials_option(SUNDIALS_TEST_UNITTESTS BOOL
   "Include unit tests in make test" OFF ADVANCED)
+
+sundials_option(ENABLE_CUSTOM_MPI_RUN_COMMAND BOOL "Enable passing in a custom command prefix to run when running MPI jobs" OFF ADVANCED)
+
+sundials_option(CUSTOM_MPI_RUN_COMMAND_PREFIX STRING "Executable prefix to append before each MPI job command." "flux mini run" ADVANCED)
```

```suggestion
sundials_option(SUNDIALS_TEST_MPIRUN_COMMAND STRING "Job scheduler or mpirun command used to launch SUNDIALS MPI tests" "${MPIEXEC_EXECUTABLE}" ADVANCED)
```

I think we can always use this variable in place of `MPIEXEC_EXECUTABLE` and just have it default to `MPIEXEC_EXECUTABLE`.  Then we do not need the `ENABLE_CUSTOM_MPI_RUN_COMMAND` option. 

On Corona, if you were to run cmake manually then you would do:
```
cmake -DSUNDIALS_TEST_MPIRUN_COMMAND="flux mini alloc"
```

To make this work in the CI, what you will need to do is add a spack variant that allows to set this value to the sundials spack package in `.gitlab/spack_packages/sundials/package.py`. We can discuss how to do this further offline if need be.